http/handler.go

```diff
@@ -276,10 +276,14 @@ func wrapGenericHandler(core *vault.Core, h http.Handler, props *vault.HandlerPr
 		} else {
 			ctx, cancelFunc = context.WithTimeout(ctx, maxRequestDuration)
 		}
+
+		defer cancelFunc()
```

Comment:
I believe not using defer in here was on purpose since this is a hot-path and `defer` adds a bit of overhead (albeit this has been greatly reduced in recent Go releases).

Reply:
I'll rever the `defer`.